src/request_handlers/shop.rs

```diff
@@ -40,19 +43,24 @@ enum ShopRequest {
 #[serde(tag = "type")]
 pub enum ShopResponse {
     DownloadOkay {
-        id: i32,
+        /// app_id of the downloaded app.
+        app_id: String,
         /// Name to be used as filename in `sendToChat`.
         name: String,
         /// Base64 encoded webxdc.
         data: String,
     },
     DownloadError {
-        id: i32,
+        app_id: String,
         error: String,
     },
     Update {
+        /// List of new app ids.
         app_infos: Vec<AppInfo>,
```

Comment:
Isn't an AppInfo an app_id/version so this is not only about new apps (app_ids) but about any new version of any app?

Reply:
This field is only used by the frontend to set every app needing an update to `AppState.Updating`.
It notifies the frontend of which apps it will be sent from the backend in the upcoming messages.

Reply:
If we want to add a version field, the we would have to put them into `DownloadOkay`.

Reply:
I updated the comments about this one.